tools/isopack.js

```diff
@@ -1100,18 +1100,64 @@ _.extend(Isopack.prototype, {
         !f.match(/^examples\/unfinished/);
     });
 
-    var gitSha = files.runGitInCheckout('rev-parse', 'HEAD');
-
+    // Regexes matching paths to transpile using babel
+    var transpileRegexes = [
+      /^tools\/[^\/]+\.js$/, // General tools files
+      // We don't support running self-test from an install anymore
+    ];
+
+    // Split pathsToCopy into two arrays - one of files that should be copied
+    // directly, and one of files that should be transpiled with Babel
+    var pathsToTranspile = [];
+    var pathsToCopyStraight = [];
+    pathsToCopy.forEach((path) => {
+      var shouldTranspile =
+        _.some(transpileRegexes, (regex) => path.match(regex));
+
+      if (shouldTranspile) {
+        pathsToTranspile.push(path);
+      } else {
+        pathsToCopyStraight.push(path);
+      }
+    });
 
+    // Set up builder to write to the correct directory
     var toolPath = 'mt-' + archinfo.host();
     builder = builder.enter(toolPath);
+
+    // Transpile the files we selected
+    var babel = require("babel");
+    pathsToTranspile.forEach((path) => {
+      var fullPath = files.convertToOSPath(
+        files.pathJoin(files.getCurrentToolsDir(), path));
+
+      var inputFileContents = files.readFile(fullPath, "utf-8");
+
+      // #RemoveInProd
+      // We don't actually want to load the babel auto-transpiler when we are
+      // in a Meteor installation where everything is already transpiled for us.
+      // Therefore, strip out that line in main.js
+      if (path === "tools/main-transpile-wrapper.js") {
+        inputFileContents = inputFileContents.replace(/.+#RemoveInProd.+/, "");
+      }
+
+      var transpiled = babel.transform(inputFileContents, _.extend({
+        filename: path
+      }, require('./babel-config.js')));
```

Comment:
It seems like it would not be difficult to write the source map to a separate file here, rather than inlining it, which might improve startup time for the tool, since source maps are usually 1.5-2x larger than source files.